apps/runtime/src/main/java/quarks/apps/runtime/MonitorApp.java

```diff
@@ -0,0 +1,204 @@
+        try {
+            Set<ApplicationServiceMXBean> controls = 
+                    controlService.getControls(ApplicationServiceMXBean.class);
+            if (controls.isEmpty()) {
+                throw new IllegalStateException(
+                        "Could not find a registered control with the following interface: " + 
+                        ApplicationServiceMXBean.class.getName());                
+            }
+            for (ApplicationServiceMXBean control : controls)
+// TODO add ability to submit with the initial application configuration
+                control.submit(applicationName, null);
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+    
+    /**
+     * Declares the following topology:
+     * <pre>
+     * JobEvents source --> Filter (health == unhealthy) --> Restart application
+     * </pre>
+     * 
+     * @param name the topology name
+     * @return the application topology
+     */
+    protected Topology declareTopology(String name) {
+        Topology t = provider.newTopology(name);
+        TStream<JsonObject> jobEvents = JobEvents.source(
+                t, 
+                (evType, job) -> { return MonitorAppEvent.toJsonObject(evType, job); }
+                );
+        jobEvents = PlumbingStreams.isolate(jobEvents, true);
+
+        jobEvents = jobEvents.filter(
+                value -> {
+                    logger.trace("Filter: {}", value);
+
+                    try {
+                        JsonObject job = MonitorAppEvent.getJob(value);
+                        return (Job.Health.UNHEALTHY.name().equals(
+                                MonitorAppEvent.getJobHealth(job)));
+                    } catch (IllegalArgumentException e) {
+                        logger.info("Invalid event filtered out, cause: {}", e.getMessage());
+                        return false;
+                    }
+                 });
+
+        jobEvents.sink(new JobRestarter(t.getRuntimeServiceSupplier()));
+        return t;
+    }
+
+    /**
+     * A {@code Consumer} which restarts the application specified by a 
+     * JSON object passed to its {@code accept} function. 
+     */
+    private static class JobRestarter implements Consumer<JsonObject> {
+        private static final long serialVersionUID = 1L;
+        private final Supplier<RuntimeServices> rts;
+
+        JobRestarter(Supplier<RuntimeServices> rts) {
+            this.rts = rts;
+        }
+
+        @Override
+        public void accept(JsonObject value) {
+            ControlService controlService = rts.get().getService(ControlService.class);
+            JsonObject job = MonitorAppEvent.getJob(value);
+            String applicationName = MonitorAppEvent.getJobName(job);
+
+            logger.info("Will restart monitored application {}, cause: {}", applicationName, value);
+            submitApplication(MonitorAppEvent.getJobName(job), controlService);
```

Comment:
replace 2nd call to `getJobName()` with `applicationName`?  Not an efficiency concern, just makes things clearer.